src/Microsoft.AspNetCore.Razor.Language/Syntax/InternalSyntaxList.cs

```diff
@@ -0,0 +1,388 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal abstract class InternalSyntaxList : GreenNode
+    {
+        internal InternalSyntaxList()
```

Comment:
All these `Internal***` types work on green nodes. I named them this way to differentiate from similar types that operate on red nodes.

Reply:
Sounds good to me.